app/src/main/java/com/github/se/project/model/lesson/LessonViewModel.kt

```diff
@@ -31,49 +36,55 @@ class LessonsViewModel(private val repository: LessonRepository) : ViewModel() {
         object : ViewModelProvider.Factory {
           @Suppress("UNCHECKED_CAST")
           override fun <T : ViewModel> create(modelClass: Class<T>): T {
-            return LessonsViewModel(LessonRepositoryFirestore(Firebase.firestore)) as T
+            return LessonViewModel(LessonRepositoryFirestore(Firebase.firestore)) as T
           }
         }
   }
 
   /**
-   * Retrieves all lessons associated with a specific user.
+   * Generates a new unique ID.
    *
-   * @param userUid The UID of the user whose lessons are to be retrieved.
+   * @return A new unique ID.
    */
-  fun getLessonsByUser(userUid: String) {
-    repository.getLessonsByUserId(
-        userUid,
-        onSuccess = { lessons_.value = it },
-        onFailure = { e -> Log.e("LessonViewModel", "Error loading user: $userUid's lessons", e) })
+  fun getNewUid(): String {
+    return repository.getNewUid()
   }
 
   /**
    * Adds a new lesson to the repository.
    *
    * @param userUid The UID of the user to associate with the lesson.
    * @param lesson The Lesson object to be added.
+   * @param onComplete Callback to execute when the operation completes.
    */
-  fun addLesson(userUid: String, lesson: Lesson) {
-    repository.addLessonByUserId(
-        userUid = userUid,
+  fun addLesson(userUid: String, lesson: Lesson, onComplete: () -> Unit) {
+    repository.addLesson(
         lesson = lesson,
-        onSuccess = { getLessonsByUser(userUid) }, // Refresh the lesson list on success
-        onFailure = { Log.e("LessonViewModel", "Error adding lesson: $lesson", it) })
+        onSuccess = {
+          onComplete() // Call the provided callback on success
+        },
+        onFailure = {
+          Log.e("LessonViewModel", "Error adding lesson: $lesson", it)
+          onComplete() // Call the callback even if there's a failure
```

Comment:
If onComplete() is called both for success and Failure, to avoid dupplication we can put it after the repository.add(...) call (this is very a detail and it has not a lot of importance).